### PR TITLE
Merge from Cva 0.4

### PR DESCRIPTION
**Pull request by priesgo**
* Created a development version for every model named `${version}-SNAPSHOT`.
* Build script now controls the return code of system commands and fails and logs errors if necessary.
* Some model changes in CommonInterpretedCancer.avdl to support cancer cases in CVA.
* A lot of CVA model changes...

@Antonior26 we need to review the changes for ReportedVariantCancer. We have added an entity called `VariantCall` that will be homologous to `CalledGenotype`. We also added a `groupOfVariants` to the ReportEventCancer.

### CODE REVIEW

**Comment by greglever** [comment]
@priesgo how does this affect how we build the models ?
Can we still just do: 
```
    mvn clean install -DskipTests && \
    mvn clean generate-sources && \
    python build.py
```
after we have the rest of the dependencies ?

**Comment by priesgo** [comment]
So, `build.py` is now controlling better the return codes of the system calls and it fails immediately when any step fails. Before it would just continue running and you will get a stack of errors and in some situations those errors will not make `build.py` execution fail. For instance, the generation of avro documentation failure could go unnoticed.
The drawback is that in the Jenkins integration if `avrodocs` dependency is not met the build will fail now, when it did not fail before. Worth double checking this.

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+   - ReportedVariantInjectRD
+   - CandidateVariantInjectRD
+
+Every ReportedVariant is uniquely identified by:
+
+   - report event id
+   - report model version
+   - id
+   - version
+   - family id
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: shall we change this to `group id` to reflect the changes in the CIP API ?

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        string id;
+        /**
+        The version for the entity containing this reported variant. This is a correlative number being the highest
+        value the latest version.
+        */
+        int version;
+        /**
+        The family identifier
+        */
+        string familyId;
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: shall we change this to `groupId` to reflect the changes to `group_id` in the CIP API ?

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        */
+        string parentId;
+        /**
+        The version for the parent entity
+        */
+        int parentVersion;
+        /**
+        The family identifier
+        */
+        string familyId;
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: shall we change this to `groupId` to reflect the changes to `group_id` in the CIP API ?

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: shall we change this to `groupId` to reflect the changes to `group_id` in the CIP API ?

**Comment by priesgo** [comment]
OK, the build error was a problem in the docker environment. I agreed with @greglever how to set it up to point to our artifactory repository that has the correct dependencies. This will be fixed after this massive PR is merged.

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl (hunk 1 above)
Comment (priesgo) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: agreed

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl (hunk 2 above)
Comment (priesgo) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: agreed

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl (hunk 3 above)
Comment (priesgo) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: agreed

File: schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl (hunk 4 above)
Comment (priesgo) [inline schemas/IDLs/org.gel.models.cva.avro/0.4.0-SNAPSHOT/DataIntakeRD.avdl]: agreed

File: resources/CodeGenerationFromGA4GH/process_schemas.py [hunk 5]
```diff
@@ -233,7 +233,7 @@ def write(self, outputFile):
         if doc is None:
             doc = "No documentation"
         self._writeWithIndent('"""', outputFile)
-        self._writeWrappedWithIndent(doc, outputFile)
+        self._writeWrappedWithIndent(re.sub(r'[^\x00-\x7F]+', ' ', doc), outputFile)
```
Comment (greglever) [inline resources/CodeGenerationFromGA4GH/process_schemas.py]: :+1:

Comment (priesgo) [inline resources/CodeGenerationFromGA4GH/process_schemas.py]: goooood

File: build.py [hunk 6]
```diff
@@ -180,13 +218,15 @@ def main():
         nargs='+',
         default=[
             'org.gel.models.participant.avro::1.0.0', 'org.gel.models.participant.avro::1.0.1',
-            'org.gel.models.participant.avro::1.0.2',
+            'org.gel.models.participant.avro::1.0.2', 'org.gel.models.participant.avro::1.0.3-SNAPSHOT',
```
Comment (priesgo) [inline build.py]: I did that

File: schemas/IDLs/org.opencb.biodata.models/1.2.0/evidence.avdl [hunk 7]
```diff
@@ -316,12 +320,12 @@ Evidence of benign impact:
...
-        SO_0001773,
-        SO_0002054,
-        SO_0001786,
-        SO_0002055
+        dominant_negative_variant,
+        gain_of_function_variant,
+        lethal_variant,
+        loss_of_function_variant,
+        loss_of_heterozygosity,
+        null_variant
```
Comment (priesgo) [inline schemas/IDLs/org.opencb.biodata.models/1.2.0/evidence.avdl]: there is one of this issues which is like the Pisuerga (i.e.: appears and disappears) with the download and extraction of avdl from biodata and the quick&dirty workaround was to commit the biodata models in the repo. 
I have to fix this!

File: setup.py [hunk 8]
```diff
@@ -15,7 +15,7 @@
     "pysam"
 ]
 
-VERSION = "4.2.0"
+VERSION = "4.3.0-SNAPSHOT"
```
Comment (priesgo) [inline setup.py]: goooood. I don't like at all but we have to keep versions synchronised both in `setup.py` and `pom.xml`

**Comment by priesgo** [comment]
Please check that it works on your side now.
```
(gelreportmodelstest) priesgo@londoner2:~/src/gel/GelReportModels$ python -m unittest discover
..........................................
----------------------------------------------------------------------
Ran 42 tests in 2.083s

OK
```

Documentation generation has been simplified, which has the advantage of not requiring to maintain a list of classes for which we want to generate docs for each package. This also has the secondary effect that independent documentation will be generated for classes like CommonInterpreted, while before it was only generated as a dependency inside other classes.

Also package names have all now dashes replaced by underscores.

**Comment by priesgo** [comment]
We may want to integrate the python unit tests in maven, but also the package setup.py install in the initialization phase.

**Pull request merged**